### DISCUSSION TITLE
AiYaTo RGBCW Bulb - Does not retain color settings?

### DISCUSSION THREAD

**Discussion by k0jdd** [arendst/Tasmota#18028] [Q&A] [answered]
I have an AiYaTo RBWCW bulb that is alway on.  I change the color of this bulb depending on the state of the alarm system. This is controlled using StartThings.

- The bulb came pre-flashed with Tasmota 12.0.2
- I suspect that AiYaTo compiles their own binaries because the build includes a template for this bulb as Module 0.

- I have read and re-read the Tasmota documentation concerning Lights.
 - I have read through all the documentation for every command.

I have two 'Issues' and I am seeking input.  

Issue 1. Out of the box the bulb does NOT preserve its' last RGB state. Meaning that if I set the color (using any method), then power cycle the bulb, it comes back as WHITE.  Smartthings has no option for me to check the color of the bulb or I would do that. 

Solution 1. I set PowerOnState 0 and then I wrote a ST routine that says any time the bulb is OFF, turn it ON and set the appropriate Color. 

This works fine, but keep reading.

Issue 2. The bulb will occasionally restart itself due to Hardware Watchdog. This is probably caused by a cheap power supply in the bulb (which I cannot control but what do you want for $5?).  When this happens the bulb powers up as ON and is WHITE. I believe that this happens because PowerOnState is not executed on a warm restart per the documentation.

Solution 2.  I added a rule that says to turn the bulb OFF on SystemBoot.

This solves the issue, but keep reading.

Analysis.

It seems to me that remembering where you were when you restarted is a pretty basic thing.  If I did not had issue 1 then issue 2 would not even exist.  

Am I missing something that I could configure that would resolve issue 1 and thus eliminate the need for any additional rules or routines or whatever?

Thanks for reading.

John

**Comment by sfromis** [comment, marked as answer]
Standard Tasmota features controlling what/when saving are `SetOption0` (controls saving of power state, not color) and `SaveData` which may be customized to only save parameter changes (including color) manually, or within a set interval after last change, default 1 second, but may be changed by them.

The case for *not* saving color setup may be to reduce flash wear, if you are making lots of changes at short intervals, and that users (including me) will actually prefer to avoid having some old color state restored. My preference is instead to have a "reasonable initial" state saved, and then to not save changes during operation. YMMV, obviously.

**Reply by k0jdd** [reply]
@sfromis 

Thanks for the reply.  

SaveData is on.  But when watching the weblog whilst changing the color I never see a flash write. (But I see them for other events). If would **_not_** be surprised if they modified the code in this respect.  

I do not disagree with any of your other analysis as to why this might be the way it is. I was just wondering if I had missed a blindingly obvious solution in the documentation.

**Reply by sfromis** [reply]
I do not know if they changed something in Tasmota behavior with that factory-flashed version. Maybe check details of configuration, using the tool decode-config. Maybe rules were added to do something. Maybe you have some automation (or retained commands) kicking in right after boot, do check the log after having enabled `Weblog 3` for details, before restarting.

**Reply by barbudor** [reply]
I did applied the template to a Nodemcu, set some lights settings and power cycled and it was recovering the values except the CT one. So I was maybe missing a configuration

May be time for a reflash and reset?
Or asked them what they changed and how they configured the bulb

Regarding reboots if this happens specially at high brightness it might indeed be because of the power supply limitations.
I think there are tasmota settings to avoid pushing all 5 leds to their maximum simultaneously but haven't found it (as I don't use such kind of bulbs)

**Reply by sfromis** [reply]
As the two AiYaTo RGBCCT bulbs available in the templates repository are pretty plain 5-PWM configs, DimmerRange can be used to cap how high Tasmota will go, if it proves to improve stability to avoid highest brightness levels.

But yeah, I'd be "strongly tempted" to switch to fully standard Tasmota to know exactly what has been done, after at least backing up the config they delivered the bulb with. However, with bulbs doing things like firmware backup, and returning to old, gets harder as it needs wired flashing for this. Thus, upgrading to standard Tasmota may be a one-way street (not that I'd worry too much).
